packages/uploadthing/src/internal/handler.ts

```diff
@@ -336,24 +336,7 @@ export const buildRequestHandler = <TRouter extends FileRouter>(
           });
         }
 
-        let callbackUrl = url;
-        if (config?.callbackUrl) {
-          callbackUrl = getFullApiUrl(config.callbackUrl);
-        } else if (process.env.UPLOADTHING_URL) {
-          callbackUrl = getFullApiUrl(process.env.UPLOADTHING_URL);
-        }
-
-        if (
-          process.env.NODE_ENV === "production" &&
-          callbackUrl.host.includes("localhost")
-        ) {
-          console.warn(
-            [
-              "[UT] [WARN] You are using a localhost callback url in production which is not supported.",
-              "Read more and learn how to fix it here: https://uploadthing.com/faq#my-callback-runs-in-development-but-not-in-production",
-            ].join(" "),
-          );
-        }
+        const callbackUrl = resolveCallbackUrl({ config, req, url });
```

Comment:
This is only used once below, can we just inline it?

Reply:
Also is there a reason we cant just pass callbackUrl in the fetch, and need to do

`callbackUrl: callbackUrl.origin + callbackUrl.pathname,` ?

Reply:
didn't wanna pass any potential search params etc to UT

Reply:
ahh makes sense.